src/transformers/models/falcon/modeling_falcon.py

```diff
@@ -564,6 +564,12 @@ def forward(
 
         past_key_value = (key_layer, value_layer) if use_cache else None
 
+        # TODO: These transpose are quite inefficient but Flash Attention requires the layout [batch_size, sequence_length, num_heads, head_dim]. We would need to refactor the KV cache
+        # to be able to avoid many of these transpose/reshape/view.
+        query_layer = query_layer.transpose(1, 2)
+        key_layer = key_layer.transpose(1, 2)
+        value_layer = value_layer.transpose(1, 2)
```

Comment:
@younesbelkada fyi Falcon with flash attention is currently bugged due to these missing.